bugfix: actually raise on too many code collisions

gumchum opened the pull request:
Creating codes for invitations and invitation requests should raise on
the 10th attempt.

The "too many hash collisions" exception is never raised because the 10th try never happens. Using the inclusive range operator, instead of the exclusive one, makes it work.

Aside: the code columns for these models are missing a unique index. Is it worth adding or are too few concurrent creates for invitations and invitation requests for that to matter?

pushcx commented:
OK, first off, how did you find this bug? If you hit it in production, either you run _by far_ the busiest instance of the codebase or you should be buying more lottery tickets.

In any case, thank you for finding, reporting, and fixing this bug. It occurs to me that this code could remove this coupling entirely by having the `(1..10)` loop set the code and `return unless Invitation.exists?(:code => self.code)`, moving the `raise` below the loop and throwing away the conditional wrapping it. Then "do this 10 times" is expressed only once and can't slip out of sync. Sound good?

gumchum commented:
I actually just found this while casually browsing the code base looking for inspiration. I've updated the code as you've suggested and removed the ranges in favor of `10.times`.

pushcx commented:
Thanks for contributing this fix back after your reading, then. :)
